swift_upload_runner/replicate.py

```diff
@@ -0,0 +1,309 @@
+                            object_name
+                        ),
+                        data=self.a_generate_object_from_reader(resp_g),
+                        headers=headers,
+                        timeout=REPL_TIMEOUT
+                ) as resp_p:
+                    if resp_p.status == 408:
+                        raise aiohttp.web.HTTPRequestTimeout()
+                    if resp_p.status not in {201, 202}:
+                        raise aiohttp.web.HTTPBadRequest(
+                            reason="Couldn't upload object segment"
+                        )
+                LOGGER.debug(f"Success in copying object {object_name}")
+            else:
+                # Ensure the segment container exists, since performing
+                # segmented upload
+                LOGGER.debug(f"Copying object {object_name} in segments.")
+                await self.a_create_container(segmented=True)
+
+                manifest = await self.a_sync_object_segments(
+                    resp_g.headers["X-Object-Manifest"]
+                )
+
+                LOGGER.debug("Uploading manifest")
+                # Add manifest headers
+                headers["X-Object-Manifest"] = manifest
+                # Create manifest file
+                async with self.client.put(
+                        common.generate_download_url(
+                            self.host,
+                            container=self.container,
+                            object_name=object_name
+                        ),
+                        data=b"",
+                        headers=headers,
+                        timeout=REPL_TIMEOUT
+                ) as resp:
+                    if resp.status != 201:
+                        raise aiohttp.web.HTTPInternalServerError(
+                            reason="Manifest creation failure."
+                        )
+                LOGGER.debug(f"Uploaded manifest for {object_name}")
+
+    async def a_copy_from_container(self):
+        """Copy objects from a source container."""
+        LOGGER.debug(
+            f"Fetching objects from container {self.source_container}"
+        )
+        container_url = common.generate_download_url(
+            self.source_host,
+            container=self.source_container
+        )
+        LOGGER.debug(f"Container url: {container_url}")
+        async with self.client.get(
+                common.generate_download_url(
+                    self.source_host,
+                    container=self.source_container,
+                ),
+                headers={
+                    "X-Auth-Token": self.auth.get_token()
+                },
+                timeout=REPL_TIMEOUT
+        ) as resp:
+            if resp.status != 200:
+                LOGGER.debug(
+                    f"Container fetch failed with status {resp.status}"
+                )
+                raise aiohttp.web.HTTPBadRequest(
+                    reason="Couldn't fetch the source container"
```

Comment:
```suggestion
                    reason="Container source fetch failed."
```
This is an example of how the responses could look better. No need to change but having `Couldn't` or `Can't` might not be a good option. It seems like the message pointed in the logs is clearer.

It is a big effort to replace all now, but at some point might be a good idea to revisit this error messages - a `Good First Issue` for external contributions ?

Reply:
Ctrl+f through messages, they should now be a bit more sensible.